api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirEncounterServiceImplTest.java

```diff
@@ -70,16 +104,84 @@ public void shouldGetEncounterByUuid() {
 	}
 	
 	@Test
-	public void findEncountersByPatientIdentifier_shouldReturnCollectionOfEncounters() {
-		when(dao.findEncountersByPatientIdentifier(PATIENT_IDENTIFIER))
-		        .thenReturn(Collections.singletonList(openMrsEncounter));
+	public void searchForEncounter_shouldReturnCollectionOfEncounterByDate() {
+		Collection<Encounter> encounters = new ArrayList<>();
+		DateRangeParam dateRangeParam = new DateRangeParam(new DateParam(ENCOUNTER_DATETIME));
+		
+		encounters.add(openMrsEncounter);
+		
+		fhirEncounter.setId(ENCOUNTER_UUID);
+		when(dao.searchForEncounters(argThat(is(dateRangeParam)), any(), any(), any())).thenReturn(encounters);
 		when(encounterTranslator.toFhirResource(openMrsEncounter)).thenReturn(fhirEncounter);
 		
-		Collection<org.hl7.fhir.r4.model.Encounter> results = encounterService
-		        .findEncountersByPatientIdentifier(PATIENT_IDENTIFIER);
-		assertThat(results, notNullValue());
-		assertThat(results.size(), equalTo(1));
-		assertThat(results.stream().findFirst().isPresent(), is(true));
+		Collection<org.hl7.fhir.r4.model.Encounter> results = encounterService.searchForEncounters(dateRangeParam, null,
+		    null, null);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getId(), equalTo(ENCOUNTER_UUID));
 	}
 	
+	@Test
+	public void searchForEncounter_shouldReturnCollectionOfEncounterByLocation() {
+		Collection<Encounter> encounters = new ArrayList<>();
+		ReferenceParam location = new ReferenceParam();
```

Comment:
Ditto on ordering

Reply:
@ibacher Not sure what to do here, we haven't included any parameter like order

Reply:
@ibacher Got it you mean that the method name should be ordered logical

Reply:
I mean the order that the code is written in.

Reply:
Basically, I'd like to see something like:

```java
ReferenceParam location = new ReferenceParam();
location.setValue(ENCOUNTER_ADDRESS_STATE);
location.setChain(Location.SP_ADDRESS_CITY)

Collection<Encounter> encounters = new ArrayList<>();
encounters.add(openMrsEncounter);

fhirEncounter.setId(ENCOUNTER_UUID);
```

So that each part is created and populated in order.